community/io/src/main/java/org/neo4j/io/pagecache/impl/muninn/PageList.java

```diff
@@ -363,12 +362,22 @@ public boolean decrementUsage( long pageRef )
 
     public long getFilePageId( long pageRef )
     {
-        return UnsafeUtil.getLong( offFilePageId( pageRef ) );
+        int highByte = UnsafeUtil.getByte( offFilePageId( pageRef ) ) & UNSIGNED_BYTE_MASK;
+        long lowInt = UnsafeUtil.getInt( offFilePageId( pageRef ) + Byte.BYTES ) & UNSIGNED_INT_MASK;
+        long filePageId = (((long) highByte) << Integer.SIZE) | lowInt;
```

Comment:
Can't you replace this with `UnsafeUtil.getLong( offFilePageId( pageRef ) ) >>> 24`? This method is called as part of `isBoundTo`, which in turn is called on every pin.

Speaking of `isBoundTo`, the file page id and the swapper id are now part of the same aligned word. So could check the binding with a single compare by masking/shifting out the usage stamp, and comparing it with a reconstructed binding word. A single compare for both fields.

Reply:
talked offline about it, we will look on it separately to see if it make sense to do it